### ISSUE TITLE
Help with exporting offline dataset for imitation learning.  Getting Nan warning when using IS/WIS estimator

### ISSUE THREAD

**Issue by witsang** [ray-project/ray#10458]
rrlib version: 0.8.7 
OS: windows
tensorflow: 2.0

I am looking for advice for imitation learning from player data.  I am generating offline experience external from my game instance.  I have no idea what to set action_prob for my SampleBatch., I am capturing player inputs and generating the SampleBatch file, thus I just set action_prob to 1 

I tested my player experience data by:
Method 1. MARWIL policy
`# Off-policy: DQN (discrete actions only) and MARWIL (discrete/continous action)
# On-pllicy: PPO, etc, requires logits, vf_preds, value_target and advantages to be exported
run: MARWIL 
env: BasicBrain
config:
  env_config:
    timescale: 1
  model:
    fcnet_hiddens: [256, 256]
    ## supervised model losses over offline data
    # custom_model_config:
    #   input_files: C:/Experiences/3DBallBrain_20200828_XA5BAhQWM0e5CTUiE-NFNg # folder of experience json
    #   imitation_loss_weight: 10 # policy_loss + imitation_loss_weight * self.imitation_loss
  # Compare IL (beta=0) vs MARWIL (When beta is 0, MARWIL is reduced to imitation learning)
  beta: 0.5
  #input: /hostfs/3DBallBrain_20200828_XA5BAhQWM0e5CTUiE-NFNg
  input: C:/Experiences/3DBallBrain_20200828_XA5BAhQWM0e5CTUiE-NFNg
   # Specify how to evaluate the current policy. This only has an effect when reading offline experiences. Available options:
    #  - "wis": the weighted step-wise importance sampling estimator.
    #  - "is": the step-wise importance sampling estimator.
    #  - "simulation": run the environment in the background, but use this data for evaluation only and not for learning.
  input_evaluation: ["is","simulation"]
  # Whether to run postprocess_trajectory() on the trajectory fragments from
  # offline inputs. Note that postprocessing will be done using the *current*
  # policy, not the *behavior* policy, which is typically undesirable for on-policy algorithms.
  postprocess_inputs: True`

Note when I use the **importance sample estimator** wis or is.  I get a lot of Nan or Inf found in input tensor warning.  
The code in estimate of ImportanceSamplingEstimator, the value of p just get larger and larger as the batch count increase since it is doing pt_prev * new_prob[t] / old_prob[t], and old_prob[t]=batch["action_prob"]=1
`p = []
        for t in range(batch.count - 1):
            if t == 0:
                pt_prev = 1.0
            else:
                pt_prev = p[t - 1]
            p.append(pt_prev * new_prob[t] / old_prob[t])`
So I don't think how I should compute this action_prob when capturing from player data?

Method 2. override custom_loss in my model to add an imitation loss:  What is a reasonable imitation_loss_weight to use and how to know if it is imitating from tensorboard graphs.
`def custom_loss(self, policy_loss, loss_inputs):
        # Create a new input reader per worker.
        self.policy_loss = policy_loss
        if self.has_input_files:
            custom_model_config = self.model_config["custom_model_config"]
            imitation_loss_weight = 10
            if 'imitation_loss_weight' in custom_model_config:
                imitation_loss_weight = custom_model_config['imitation_loss_weight']

            reader = JsonReader(custom_model_config["input_files"])
            input_ops = reader.tf_input_ops()
            #print(reader.next())

            # Define a secondary loss by building a graph copy with weight sharing.
            obs = restore_original_dimensions(
                tf.cast(input_ops["obs"], tf.float32), self.obs_space)
            logits, _ = self.forward({"obs": obs}, [], None)

            # You can also add self-supervised losses easily by referencing tensors
            # created during _build_layers_v2(). For example, an autoencoder-style
            # loss can be added as follows:
            # ae_loss = squared_diff(
            #     loss_inputs["obs"], Decoder(self.fcnet.last_layer))
            # print("FYI: You can also use these tensors: {}, ".format(loss_inputs))

            # Compute the IL loss.
            action_dist = self.dist_class(logits, self.model_config)
            
            self.imitation_loss = tf.reduce_mean(-action_dist.logp(input_ops["actions"]))
            # Add the imitation loss to each already calculated policy loss term.
            return policy_loss + imitation_loss_weight * self.imitation_loss
        return super().custom_loss(policy_loss, loss_inputs)
`
